core/src/dev/protection.c

```diff
@@ -24,146 +21,5 @@ uint8_t KEK[2][16] =
-  }
-
-  size_t xpub_len;
-
-  Result r = aes_gcm_decrypt(KEYID_PUBKEY_ENCRYPTION_KEY,
-                             encrypted_pub_key->encrypted_pub_key.bytes,
-                             encrypted_pub_key->encrypted_pub_key.size,
-                             (uint8_t *)xpub, XPUB_SIZE -1, &xpub_len);
-  if (Result_SUCCESS != r)
-  {
-    ERROR("%s failed (%d)", __func__, r);
-    return r;
-  }
-
-  // Null terminate xpub
-  xpub[xpub_len] = 0;
-  DEBUG("xpub: %s", xpub);
-
-  return Result_SUCCESS;
-}
-
-/**
- * Encrypt master_seed with master seed encryption key.
- *
- * Zeros master_seed.
- */
-Result protect_wallet(uint8_t master_seed[static MASTER_SEED_SIZE],
-                      EncryptedMasterSeed *encrypted_master_seed) {
-
- static_assert(sizeof(encrypted_master_seed->encrypted_master_seed.bytes) >=
-                    MASTER_SEED_SIZE,
-                "misconfigured encrypted_master_seed max size");
-
-  size_t ciphertext_len;
-  Result r = aes_gcm_encrypt(
-      KEYID_MASTER_SEED_ENCRYPTION_KEY, master_seed, MASTER_SEED_SIZE,
-      encrypted_master_seed->encrypted_master_seed.bytes,
-      sizeof(encrypted_master_seed->encrypted_master_seed.bytes),
-      &ciphertext_len);
-  if (r != Result_SUCCESS) {
-    ERROR("aes_gcm_encrypt failed (%d).", r);
-    return r;
-  }
-  encrypted_master_seed->encrypted_master_seed.size = ciphertext_len;
-  DEBUG("ciphertext_len: %zu", ciphertext_len);
-
-  DEBUG_("ciphertext: ");
-  for (unsigned int i = 0; i < ciphertext_len; i++) {
-    DEBUG_("%02x", encrypted_master_seed->encrypted_master_seed.bytes[i]);
-  }
-  DEBUG_("\n");
-
-  return Result_SUCCESS;
-}
-
-/**
- * Decrypt master_seed with master seed encryption key.
- */
-Result expose_wallet(EncryptedMasterSeed *encrypted_master_seed,
-                     uint8_t master_seed[static MASTER_SEED_SIZE]) {
-
-  if (encrypted_master_seed->encrypted_master_seed.size !=
-      MASTER_SEED_SIZE + 16 + 12) {
-    ERROR("Unexpected encrypted_master_seed size");
-    return Result_EXPOSE_WALLET_UNEXPECTED_ENCRYPTED_MASTER_SEED_SIZE_FAILURE;
-  }
-
-  size_t master_seed_len;
-  Result r = aes_gcm_decrypt(KEYID_MASTER_SEED_ENCRYPTION_KEY,
-                             encrypted_master_seed->encrypted_master_seed.bytes,
-                             encrypted_master_seed->encrypted_master_seed.size,
-                             master_seed, MASTER_SEED_SIZE, &master_seed_len);
-  if (r != Result_SUCCESS) {
-    ERROR("aes_gcm_decrypt failed (%d).", r);
-    return r;
-  }
-  if (master_seed_len != MASTER_SEED_SIZE) {
-    ERROR("Unexpected master_seed_len.");
-    return Result_EXPOSE_WALLET_UNEXPECTED_MASTER_SEED_LEN_FAILURE;
-  }
-
-  DEBUG_("master_seed: ");
-  for (unsigned int i = 0; i < master_seed_len; i++) {
-    DEBUG_("%02x", master_seed[i]);
-  }
-  DEBUG_("\n");
-
-  return Result_SUCCESS;
-}
-
+// The following implementation is shared between dev and ncipher targets
+#include "../protect.c"
```

Comment:
It's not a common patter in this codebase, but seems the most suitable to me in this case for this purpose. An alternative is to hide target specific code behind macros in `core/src/protect.c`. However, this would break the (unwritten) convention that source files direct under `core/src` do not have target-specific macros. I like the current approach, also because it does not require build script changes. We need to make sure that `protect.c` is not built and linked during the build process.